docs/src/test/java/docs/http/javadsl/server/testkit/PlayRoutesComparisonTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.http.javadsl.server.testkit;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.server.AllDirectives;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.javadsl.testkit.TestRoute;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static akka.http.javadsl.server.PathMatchers.*;
+import static akka.http.javadsl.server.Directives.*;
+
+
+public class PlayRoutesComparisonTest extends JUnitRouteTest {
+
+    static class Clients {
+        static String list() {
+            return "clientA,clientB,clientC";
+        }
+        static String get(long id) {
+            return "clientB";
+        }
+    }
+
+    static String download(List<String> names) {
+        return "images/logo.png: file contents";
+    }
+
+    static String getPage(String page) {
+        return "The requested [" + page + "].";
+    }
+
+    static String apiList(Optional<String> version) {
+        return version.map(v -> "aa,bb,cc").orElse("ff");
+    }
+
+    static String apiItems(List<String> items) {
+        return items.stream().collect(Collectors.joining(","));
+    }
+
+    static class Routes extends AllDirectives {
+
+        Route clientsAll() {
+            return
+                    // #fixed
+                    get(() -> path(segment("clients").slash("all"), () -> complete(Clients.list())));
```

Comment:
<img width="918" alt="Screenshot 2020-04-17 at 16 43 59" src="https://user-images.githubusercontent.com/762126/79582053-89fd0b80-80cb-11ea-9964-6fde09ee6481.png">


I found all these one-liners (in both tests and main code for Java) hard to read. I think that, even if it's not java idiomatic and feels for scala-esque, somthing like the following is more readable:

```suggestion
  get(() -> 
    path(segment("clients").slash("all"), 
      () -> complete(Clients.list())
    )
  );
```

I mean, if we're writing these docs for someone coming from Play, I think it's visually easier to parse this format then the one-liners and gently introduces the reader to a slightly different format.

Reply:
Reformatted.